src/Cli/dotnet/commands/dotnet-test/Program.cs

```diff
@@ -110,17 +149,67 @@ public static int Run(ParseResult parseResult)
             try
             {
                 Environment.SetEnvironmentVariable(NodeWindowEnvironmentName, "1");
-                return FromParseResult(parseResult, settings).Execute();
+                int exitCode = FromParseResult(parseResult, settings, testSessionCorrelationId).Execute();
+
+                // We run post processing also if execution is failed for possible partial successful result to post process.
+                exitCode |= RunArtifactPostProcessingIfNeeded(testSessionCorrelationId, parseResult, FeatureFlag.Default);
+
+                return exitCode;
             }
             finally
             {
                 Environment.SetEnvironmentVariable(NodeWindowEnvironmentName, previousNodeWindowSetting);
             }
         }
 
+        internal static int RunArtifactPostProcessingIfNeeded(string testSessionCorrelationId, ParseResult parseResult, FeatureFlag featureFlag)
+        {
+            if (!featureFlag.IsEnabled(FeatureFlag.ARTIFACTS_POSTPROCESSING))
+            {
+                return 0;
+            }
+
+            // VSTest runner will save artifacts inside a temp folder if needed.
+            string expectedArtifactDirectory = Path.Combine(Path.GetTempPath(), testSessionCorrelationId);
+            if (!Directory.Exists(expectedArtifactDirectory))
+            {
+                VSTestTrace.WriteTrace("No artifact found, post-processing won't run.");
+                return 0;
+            }
+
+            VSTestTrace.WriteTrace($"Artifacts found inside '{expectedArtifactDirectory}', running post-processing.");
```

Comment:
It feels weird to say we found artefacts in the in the folder while we only checked that folder exists. So if that's empty the message is a little weird.

Reply:
The folder is created by vstest side and is created only if we have artifacts. Cannot be empty...or can be empty if there's a bug.